api/contrib/envoy/extensions/compression/qatzstd/compressor/v3alpha/qatzstd.proto

```diff
@@ -0,0 +1,69 @@
+syntax = "proto3";
+
+package envoy.extensions.compression.qatzstd.compressor.v3alpha;
+
+import "google/protobuf/wrappers.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.compression.qatzstd.compressor.v3alpha";
+option java_outer_classname = "QatzstdProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/contrib/envoy/extensions/compression/qatzstd/compressor/v3alpha";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Qatzstd Compressor]
+// Qatzstd :ref:`configuration overview <config_qatzstd>`.
+// [#extension: envoy.compression.qatzstd.compressor]
+
+// [#next-free-field: 8]
+message Qatzstd {
+  // Reference to http://facebook.github.io/zstd/zstd_manual.html
+  enum Strategy {
+    DEFAULT = 0;
+    FAST = 1;
+    DFAST = 2;
+    GREEDY = 3;
+    LAZY = 4;
+    LAZY2 = 5;
+    BTLAZY2 = 6;
+    BTOPT = 7;
+    BTULTRA = 8;
+    BTULTRA2 = 9;
+  }
+
+  // Set compression parameters according to pre-defined compression level table.
+  // Note that exact compression parameters are dynamically determined,
+  // depending on both compression level and source content size (when known).
+  // Value 0 means default, and default level is 3.
+  //
+  // Setting a level does not automatically set all other compression parameters
+  // to default. Setting this will however eventually dynamically impact the compression
+  // parameters which have not been manually set. The manually set
+  // ones will 'stick'.
+  google.protobuf.UInt32Value compression_level = 1 [(validate.rules).uint32 = {lte: 22 gte: 1}];
```

Comment:
Sorry, I also saw the zstd manual said, it will support negative level: `The library also offers negative compression levels, which extend the range of speed vs. ratio preferences.` https://facebook.github.io/zstd/zstd_manual.html

I also saw the zstd envoy extension doesn't have this limit. 

So should we add this limitation? or this is the QAT implement's limitation?

Reply:
I think the compression level for 1-22 is more commonly used, so this limitation could be preserved.
And zstd envoy extension still uses `UInt32Value` that means the negative compression levels are not considered.

Reply:
Will qatzstd work with compression level greater than 22? if yes, I will prefer to keep the same api with zstd ext.

Reply:
No, qatzstd works with same compression levels as zstd.